src/pybind/mgr/cephadm/tests/test_spec.py

```diff
@@ -288,3 +305,148 @@ def test_rgw_service_name(spec: RGWSpec, dd: DaemonDescription, valid):
         with pytest.raises(OrchestratorError):
             dd.service_name()
 
+
+@pytest.mark.parametrize("spec,dd,valid",
+[
+    # https://tracker.ceph.com/issues/45293
+    (
+        ServiceSpec(
+            service_type='mds',
+            service_id="a",
+        ),
+        DaemonDescription(
+            daemon_type='mds',
+            daemon_id="a.host1.abc123",
+            hostname="host1",
+        ),
+        True
+    ),
+    (
+        # '.' char in service_id
+        ServiceSpec(
+            service_type='mds',
+            service_id="a.b.c",
+        ),
+        DaemonDescription(
+            daemon_type='mds',
+            daemon_id="a.b.c.host1.abc123",
+            hostname="host1",
+        ),
+        True
+    ),
+])
+def test_mds_service_name(spec: ServiceSpec, dd: DaemonDescription, valid):
+    if valid:
+        assert spec.service_name() == dd.service_name()
+    else:
+        with pytest.raises(OrchestratorError):
+            dd.service_name()
+
```

Comment:
The test function is the same for all test cases.  what about simply adding those to `test_rgw_service_name` while renaming it to `test_daemon_description_service_name`?